### PR TITLE
Save modal component for explore v2

### PR DESCRIPTION
**Pull request by vera-liu**
Done:
 - Added a SaveModal component, triggered by save button in QueryAndSaveButton
 - url params for saveOrOverwriteSlice stay the same as in v1 
 - when user updates control panel and save the slice, new form_data will be saved into the slice
 - Added specs for SaveModal
 - This PR includes changes from [https://github.com/airbnb/superset/pull/1610](url) since datasource_name is required in saveOrOverwriteSlice url
screen shot 2016-11-16 at 9 22 14 am

Todo:
 - add Alert to explore v2 for all ajax calls and exception handling

needs-review @mistercrunch @ascott

### CODE REVIEW

File: superset/assets/javascripts/explorev2/components/SaveModal.js [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+
+  setSaveDashboard(opt) {
+    this.setState({ save_to_dashboard_id: opt.value });
+    $('#add_to_dash_existing').prop('checked', true);
+    $('.gotodash').removeAttr('disabled');
+  }
+
+  fetchDashboards() {
+    const url = '/dashboardmodelviewasync/api/read?_flt_0_owners=' + this.props.user_id;
+    $.get(url, function (data) {
```
Comment (ascott) [inline superset/assets/javascripts/explorev2/components/SaveModal.js]: looks like we're duplicating this `$.get` here as well: https://github.com/airbnb/superset/pull/1612/files#diff-cb97812fa1c826050e9d4b6d69438938R119

we should move this fetching logic to actions file.

Comment (vera-liu) [inline superset/assets/javascripts/explorev2/components/SaveModal.js]: do we want to keep dashboards in global store? right now we only use them in SaveModal, I'm not sure if we would use them in other components.

Comment (ascott) [inline superset/assets/javascripts/explorev2/components/SaveModal.js]: we may not use this action in other components yet, but it's great to keep all the fetching logic in one place and keep the components leaner. could be an opportunity to start an actions file for dashboard actions, but i think we could put this fetch in the explore actions for now.

File: superset/assets/javascripts/explorev2/actions/exploreActions.js [hunk 2]
```diff
@@ -82,6 +79,30 @@ export function changeFilterValue(filter, value) {
   return { type: CHANGE_FILTER_VALUE, filter, value };
 }
 
-export function setFieldValue(key, value) {
-  return { type: SET_FIELD_VALUE, key, value };
+export function setFieldValue(key, value, label) {
+  return { type: SET_FIELD_VALUE, key, value, label };
+}
+
+export const SAVE_SLICE_SUCCEEDED = 'SAVE_SLICE_SUCCEEDED';
```
Comment (ascott) [inline superset/assets/javascripts/explorev2/actions/exploreActions.js]: do we need `SAVE_SLICE_STARTED` to show feedback to the user while we are saving?

File: superset/assets/javascripts/explorev2/components/SaveModal.js [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+  form_data: PropTypes.object,
+  datasource_type: PropTypes.string.isRequired,
+  user_id: PropTypes.string.isRequired,
+};
+
+class SaveModal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      save_to_dashboard_id: null,
```
Comment (ascott) [inline superset/assets/javascripts/explorev2/components/SaveModal.js]: can this be `camelCase`?

**Comment by ascott** [comment]
would love to see all async/fetching logic in an actions file and would love to see less jquery dom manipulation in favor of react patterns in the save modal component. otherwise, looking good!

**Comment by vera-liu** [comment]
Done:
 - moved ajax to action creator
 - more react state instead of jquery
 - added alerts

PTAL @ascott

File: superset/assets/javascripts/explorev2/components/SaveModal.js [hunk 4]
```diff
@@ -0,0 +1,249 @@
+/* eslint camel-case: 0 */
+import React, { PropTypes } from 'react';
+const $ = window.$ = require('jquery');
```
Comment (ascott) [inline superset/assets/javascripts/explorev2/components/SaveModal.js]: can we use `import $ from 'jquery';` here? don't think we need `window.$` in this case.

File: superset/assets/javascripts/explorev2/components/SaveModal.js [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+            <Alert>
+              {this.state.alert ? this.state.alert : this.props.alert}
+              <i
+                className="fa fa-close pull-right"
+                onClick={this.removeAlert.bind(this)}
+                style={{ cursor: 'pointer' }}
+              />
+            </Alert>
+          }
+          <form>
```
Comment (ascott) [inline superset/assets/javascripts/explorev2/components/SaveModal.js]: would be great to use the react-bootstrap form components here: https://react-bootstrap.github.io/components.html#forms

**Comment by ascott** [comment]
yes! this looks so much better! it'll be easier to maintain/iterate on in the future!

just one comment about using react-bootstrap forms, could do that in this PR or another, otherwise LGTM!

**Pull request merged**